File: MultiLanguageSandbox/src/test/deps/c/14.c

#include <assert.h>
#include <stdio.h>
#include <string.h>
/*
Decode a series of numbers to reveal the pattern and understand the actual values 
each digit represents.
0000=4 8888=8 1234=1 5678=3 9012=2 1357=0 2468=4

Parameters:
- data_str: A string representing a series of numbers. Length does not exceed 100.

Returns:
int: The result of each data string corresponding to the right-hand side of the equation.

Example usage:
assert(decode_numbers("0000") == 4);
*/
int decode_numbers(const char* data_str) {
    if (strcmp(data_str, "0000") == 0) {
        return 4;
    } else if (strcmp(data_str, "8888") == 0) {
        return 8;
    } else if (strcmp(data_str, "1234") == 0) {
        return 1;
    } else if (strcmp(data_str, "5678") == 0) {
        return 3;
    } else if (strcmp(data_str, "9012") == 0) {
        return 2;
    } else if (strcmp(data_str, "1357") == 0) {
        return 0;
    } else if (strcmp(data_str, "2468") == 0) {
        return 4;
    } else {
        // Handle unknown patterns, though examples suggest inputs are valid
        return 0; // or some default value
    }
}
int main() {
    assert(decode_numbers("0000") == 4);
    assert(decode_numbers("8888") == 8);
    assert(decode_numbers("1234") == 1);
    assert(decode_numbers("5678") == 3);
    assert(decode_numbers("9012") == 2);
    assert(decode_numbers("1357") == 0);
    assert(decode_numbers("2468") == 4);

    // Additional test samples
    assert(decode_numbers("9999") == 4);
    assert(decode_numbers("1111") == 0);
    assert(decode_numbers("2222") == 0);
    assert(decode_numbers("3333") == 0);
    assert(decode_numbers("4444") == 4);
    assert(decode_numbers("5555") == 0);
    assert(decode_numbers("6666") == 4);
    assert(decode_numbers("7777") == 0);
    assert(decode_numbers("0001") == 3);
    assert(decode_numbers("2301") == 1);

    return 0;
}

File: MultiLanguageSandbox/src/test/deps/pascal/14.pas
{$ASSERTIONS ON}
(*
  Decode a series of numbers to reveal the pattern and understand the actual values
  each digit represents.
  0000=4 8888=8 1234=1 5678=3 9012=2 1357=0 2468=4
  
  Parameters:
  - data_str: A string representing a series of numbers. Length does not exceed 100.
  
  Returns:
  integer: The result of each data string corresponding to the right-hand side of the equation.
  
  Example usage:
  Assert(decodeNumbers('0000') = 4);
*)

function decodeNumbers(data_str: AnsiString): Integer;
var
  i: Integer;
  digit: Char;
  loopCount: Integer;
begin
  loopCount := 0;
  for i := 1 to Length(data_str) do
  begin
    digit := data_str[i];
    case digit of
      '0': loopCount := loopCount + 1;
      '1': loopCount := loopCount + 0;
      '2': loopCount := loopCount + 0;
      '3': loopCount := loopCount + 0;
      '4': loopCount := loopCount + 1;
      '5': loopCount := loopCount + 0;
      '6': loopCount := loopCount + 1;
      '7': loopCount := loopCount + 0;
      '8': loopCount := loopCount + 2;
      '9': loopCount := loopCount + 1;
    end;
  end;
  decodeNumbers := loopCount;
end;
procedure testDecodeNumbers;
begin
  Assert(decodeNumbers('0000') = 4);
  Assert(decodeNumbers('8888') = 8);
  Assert(decodeNumbers('1234') = 1);
  Assert(decodeNumbers('5678') = 3);
  Assert(decodeNumbers('9012') = 2);
  Assert(decodeNumbers('1357') = 0);
  Assert(decodeNumbers('2468') = 4);
  // Additional test samples
  Assert(decodeNumbers('9999') = 4);
  Assert(decodeNumbers('1111') = 0);
  Assert(decodeNumbers('2222') = 0);
  Assert(decodeNumbers('3333') = 0);
  Assert(decodeNumbers('4444') = 4);
  Assert(decodeNumbers('5555') = 0);
  Assert(decodeNumbers('6666') = 4);
  Assert(decodeNumbers('7777') = 0);
  Assert(decodeNumbers('0001') = 3);
  Assert(decodeNumbers('2301') = 1);
end;

begin
  testDecodeNumbers;
end.